Subcatchment masking (#885)
Deltares/Wflow.jl: docs/user_guide/faq.qmd

river routing. This can lead to unrealistically large water depths when the bankfull
capacity is exceeded as there is no exchange of water between the river channel and (1D or
2D) floodplains.

### How do I run only a part of my model?
You can easily run only a subsection of your model, by modifying the
`subbasin_location__count` to a different layer in your staticmaps. By default, this layer
represents your entire catchment, and all cells that have a value are interpreted as active
cells. You can update this to another layer which contains a subset of the original data.

> Alternatively, you can point this to a layer with subcatchment information (e.g. based on

Reviewer: ```suggestion
Alternatively, you can point `subbasin_location__count` to a layer with subcatchment information (e.g. based on
```

> Alternatively, you can also point this to a layer with subcatchment information (e.g. based

Reviewer 2: ```suggestion
Alternatively, you can point this to a layer with subcatchment information (e.g. based
```

gauges with discharge observations), and use the `subbasin_active_area__count` setting to
provide a list of IDs within that layer that indicate the basins that you want to simulate.
See below for an example:

```toml
[input]
subbasin_location__count = "subcatchment_gauges_obs"

> subbasin_active_area__count = [1016, 1013, 1011, 206, 203, 503]

Reviewer: This is more in line with the already used `subbasin_location__count` entry:
```suggestion
subbasin_active_location__count = [1016, 1013, 1011, 206, 203, 503]
```

```

::: {.callout-important}

> When using this functionality, please take the following points into account:

Reviewer: Just wondering: any considerations needed when running with multiple threads/this works fine with this functionality?

Author: Not in particular; since the new outlets are assumed pits, the multithreading recognizes them as individual basins

- Ensure that all upstream regions are included (unless you make use of external inflows)

> - When using water demand and allocation: ensure that there are no allocations area's with

Reviewer 2: ```suggestion
- When using water demand and allocation: ensure that there are no allocation areas with
```

>   the same IDs outside your active domain (a warning is shown in the logging)

Reviewer 2: ```suggestion
  the same ID outside your active domain (a warning is shown in the logging)
```

- When using local_inertial for routing: a ghost node will be added at the end of each river
  section (with a set distance and water level). This might cause some difference in the
  results, as it no longer contains the "real" downstream boundary condition. If this is

>   important, ensure that the gauges for evaluation have sufficient downstream pixels in your

Reviewer 2: Does pixel here refer to a cell? I feel that's a better word to refer to the concept

Author: Thanks, yes cell is indeed a better word!

>   selected model domain.

Reviewer: Is it not better (easier) to recommend to add boundary conditions at these points?

Author: yes true, added that as well!

- When writing output at specific cells (`index`) and/or x-y positions (`coordinate`): this

>   can cause unreliable results and/or give errors. If you need output at specific locations,

Reviewer: I guess this means a check is not performed for `index` and `coordinate` (removing entries that are not part of active indices)? If that's the case maybe add this at the start. I think the "this can cause unreliable results" is a bit unclear.

Author: This was indeed not checked. I adjusted the code to show a warning rather than an error, which means that this message is no longer relevant, and users are a bit more flexible with using the same toml file

  use the `map` functionality instead, as this will still write output at the targeted
  locations when running only a subsection of your model.

> :::

Reviewer: Probably good to add what a user needs to take into account for the Darcy based groundwater flow concept?

Author: Added!
